p11-kit/proxy.c

```diff
@@ -187,12 +188,30 @@ map_session_to_real (Proxy *px,
 	return rv;
 }
 
+static CK_RV
+map_session_to_real3 (Proxy *px,
+                      CK_SESSION_HANDLE_PTR handle,
+                      Mapping *mapping,
+                      Session *session)
+{
+	CK_RV rv;
+
+	rv = map_session_to_real (px, handle, mapping, session);
+	if (rv != CKR_OK)
+		return rv;
+
+	if (mapping->funcs->version.major < 3)
+		return CKR_FUNCTION_NOT_SUPPORTED;
```

Comment:
I think I wanted to avoid writing this check in each and every new function as each of them operates on the session and this function already contained some of the sanity checks. But if you prefer, I can copy this around and remove this function.